conans/client/installer.py

```diff
@@ -404,6 +406,8 @@ def _build(self, nodes_by_level, keep_build, root_node, graph_info, remotes, bui
                         self._binaries_analyzer.reevaluate_node(node, remotes, build_mode, update)
                     _handle_system_requirements(conan_file, node.pref, self._cache, output)
                     self._handle_node_cache(node, keep_build, processed_package_refs, remotes)
+                # After a node has been managed, better reset its transitive info
```

Comment:
Is this necessary? Why is better? I'd like to think that the requirements are not modified after they are computed...

Reply:
The problem is that when upstream dependencies are built, they get a new PREV, and you need to update it in the downstream consumers, even if they haven't been built, otherwise the ``package_id()`` will not have them updated, and it will fail computing the package_id, because an upstream package has still PREV_UNKNOWN, even if it was built.